AttitudeManager/Src/MovingAverage.cpp

```diff
@@ -0,0 +1,41 @@
+#include "MovingAverage.hpp"
+#include <string.h>
+
+template<typename T>
+Window<T>::Window() {
+  memset(window, 0, MAX_WINDOW_SIZE);
+  size = 0;
+  pos = 0;
+}
+
+template<typename T>
+void Window<T>::update(T newValue) {
+  window[pos] = newValue;
+  pos++;
```

Comment:
you can post-increment within your array access I believe:

https://stackoverflow.com/questions/34250975/increment-operator-inside-array